upload/admin/controller/extension/payment/coingate.php

```diff
@@ -2,110 +2,156 @@
-      'payment_coingate_order_status_id', 'payment_coingate_pending_status_id', 'payment_coingate_confirming_status_id', 'payment_coingate_paid_status_id',
-      'payment_coingate_invalid_status_id', 'payment_coingate_expired_status_id', 'payment_coingate_canceled_status_id', 'payment_coingate_refunded_status_id',
-      'payment_coingate_total', 'payment_coingate_geo_zone_id', 'payment_coingate_receive_currency', 'payment_coingate_test_mode');
-
-
-    foreach ($fields as $field) {
-      if (isset($this->request->post[$field])) {
-  			$data[$field] = $this->request->post[$field];
-  		} else {
-  			$data[$field] = $this->config->get($field);
-  		}
+    private function getFields()
+    {
+        return [
+            'payment_coingate_status',
+            'payment_coingate_api_auth_token',
+            'payment_coingate_api_secret',
+            'payment_coingate_order_status_id',
+            'payment_coingate_pending_status_id',
+            'payment_coingate_confirming_status_id',
+            'payment_coingate_paid_status_id',
+            'payment_coingate_invalid_status_id',
+            'payment_coingate_expired_status_id',
+            'payment_coingate_canceled_status_id',
+            'payment_coingate_refunded_status_id',
+            'payment_coingate_total',
+            'payment_coingate_geo_zone_id',
+            'payment_coingate_receive_currency',
+            'payment_coingate_test_mode',
+            'payment_coingate_prefill_coingate_invoice_email'
+        ];
     }
 
-    $data['payment_coingate_sort_order'] = isset($this->request->post['payment_coingate_sort_order']) ?
-            $this->request->post['payment_coingate_sort_order'] :  $this->config->get('payment_coingate_sort_order');
-
-
-    if (empty($data['payment_coingate_api_auth_token']) && !empty($data['payment_coingate_api_secret']))
-        $data['payment_coingate_api_auth_token'] = $data['payment_coingate_api_secret'];
-
-    $data['header'] = $this->load->controller('common/header');
-    $data['column_left'] = $this->load->controller('common/column_left');
-    $data['footer'] = $this->load->controller('common/footer');
-
-    $this->response->setOutput($this->load->view('extension/payment/coingate', $data));
-  }
-
-  protected function validate() {
-    if (!$this->user->hasPermission('modify', 'extension/payment/coingate')) {
-      $this->error['warning'] = $this->language->get('error_permission');
+    protected function validate()
+    {
+        if (!$this->user->hasPermission('modify', 'extension/payment/coingate')) {
+            $this->error['warning'] = $this->language->get('error_permission');
+        }
+
+        if (!class_exists('CoinGate\Client')) {
+            $this->error['warning'] = $this->language->get('error_composer');
+        }
+
+        if (!$this->error) {
+            try {
+                \CoinGate\Client::setAppInfo('OpenCart', '1.0.0');
```

Comment:
dont hardcore version get it from COINGATE_OPENCART_EXTENSION_VERSIO

Reply:
Done